src/p2wsh.js

```diff
@@ -14,11 +16,140 @@
  */
 export const P2WSH = "P2WSH";
 
+/**
+ * @description provides the size of single tx input for a segwit tx (i.e. empty script)
+ * Each input field will look like:
+ * prevhash (32 bytes) + prevIndex (4) + scriptsig (1) + sequence bytes (4)
+ * @returns {Number} 41 (always 41 for segwit inputs since script sig is in witness)
+ */
+function txinSize() {
+  const PREVHASH_BYTES = 32;
+  const PREV_INDEX_BYTES = 4;
+  const SCRIPT_LENGTH_BYTES = 1
+  const SEQUENCE_BYTES = 4
+
+  return (PREVHASH_BYTES + 
+    PREV_INDEX_BYTES + 
+    SEQUENCE_BYTES + 
+    SCRIPT_LENGTH_BYTES
+  )
+}
+
+/**
+ * @description Returns the approximate size of outputs in tx. 
+ * Calculated by adding value field (8 bytes), field providing length
+ * scriptPubkey and the script pubkey itself
+ * @param {Number} [scriptPubkeySize = 34] size of script pubkey. 
+ * Defaults to 34 which is the size of a P2WSH script pubkey and the
+ * largest possible standard
+ * @returns {Number} size of tx output (default: 43)
+ */
+function txoutSize(scriptPubkeySize = 34) {
+  // per_output: value (8) + script length (1) + 
+  const VAL_BYTES = 8;
+  const scriptLengthBytes = encoding.sizeVarint(scriptPubkeySize);
+  // for P2WSH Locking script which is largest possible(34)
+  return VAL_BYTES + scriptLengthBytes + scriptPubkeySize;
+}
+
+/**
+ * @description calculates size of redeem script given n pubkeys.
+ * Calculation looks like: 
+ * OP_M (1 byte) + size of each pubkey in redeem script (OP_DATA= 1 byte * N) +
+ * pubkey size (33 bytes * N) + OP_N (1 byte) + OP_CHECKMULTISIG (1 byte)
+ *  => 1 + (1 * N) + (33 * N) + 1 + 1
+ * @param {Number} n - value of n in m-of-n for multisig script
+ * @returns {Number} 3 + 34 * N
+ */
+export function getRedeemScriptSize(n) {
+  const OP_M_BYTES = 1;
+  const OP_N_BYTES = 1;
+  const opDataBytes = n; // 1 byte per pubkey in redeem script
+  const pubkeyBytes = 33 * n;
+  const OP_CHECKMULTISIG_BYTES = 1;
+  return OP_M_BYTES + opDataBytes + pubkeyBytes + OP_N_BYTES + OP_CHECKMULTISIG_BYTES;
+}
+
+/**
+ * @description Calculates the value of a multisig witness given m-of-n values
+ * Calculation is of the following form:
+ * witness_items count (varint 1+) + null_data (1 byte) + size of each signature (1 byte * OP_M) + signatures (73 * M) +
+ * redeem script length (1 byte) + redeem script size (4 + 34 * N bytes)
+ * @param {Number} m - value of m in m-of-n for multisig script
+ * @param {Number} n - value of n in m-of-n for multisig script
+ * @returns {Number} 6 + (74 * M) + (34 * N)
+ */
+export function getWitnessSize(m, n) {
```

Comment:
This stays the same for both types of P2WSH, but would need to be adjusted for P2WPKH support.